module/extension/Director/src/Director.cpp

```diff
@@ -269,6 +269,21 @@ namespace module::extension {
                 // changed any of the tasks that are
                 // queued
                 reevaluate_group(g);
+                // Go check if this state update has changed
+                // any of the tasks that are queued
+                reevaluate_group(g);
+            });
+
+        on<Trigger<WaitDelay>, Sync<Director>, Pool<Director>, Priority::HIGH>().then(
+            "Wait Delay",
+            [this](const WaitDelay& w) {
+                // Get the provider that we are waiting on
+                auto provider = providers.at(w.provider->id);
+
+                // If the provider is still active, then we can run it
+                if (provider == provider->group.active_provider) {
+                    run_task_on_provider(provider->group.active_task, provider, RunReason::WAIT);
```

Comment:
Should this just fold in like any other task, and rather than having a special RunReason, just say the wait task was Done (so SubtaskDone run reason)